Mike/Aufgabe-2/main.js

```diff
@@ -0,0 +1,126 @@
+"use strict";
+
+window.addEventListener("load", function () {
+  let numbers1 = [];
+  let numbers2 = [];
+  let allNums = null;
+  let operator = null;
+  let isOperatorSelected = false;
+  let result = null;
+  let input = null;
+  let output = document.getElementById("p1");
+  let buttons = document.querySelectorAll("button");
+  let isInset = false;
+  let insetButton = null;
+
+  const inset = (btn) => {
+    if (!isInset)
+      btn.style.borderStyle =
+        btn.style.borderStyle !== "inset" ? "inset" : "outset";
+    insetButton = btn;
+  };
+
+  const outset = () => {
+    insetButton.style.borderStyle = "outset";
+  };
+
+  const addDecimal = () => {
+    !isOperatorSelected ? numbers1.push(".") : numbers2.push(".");
+  };
+
+  buttons.forEach((button) => {
+    button.addEventListener("click", function () {
+      input = this.innerText;
+
+      if (isOperatorSelected === true && operator === "/" && input === "0") {
+        output.innerHTML = 0;
+      } else {
+        if (input != null && isNaN(input)) {
```

Comment:
Erste Verbesserung: du brauchst input nicht auf nicht null überprüfen. es reicht input && isNaN(input)
Zweiter Tipp von mir bei String Switch-Case: Bei größeren Datenmengen, mache input zu klein oder Großbuchstaben, dann musst du nicht schauen, ob das jetzt in Groß oder Kleinbuchstaben geschrieben ist